### PR TITLE
Respect CritMultipler when doing damage calculations.

### LINKED ISSUE #27: Fix WCX_Engine_Crit
I don't see CritMultiplier actually being used anywhere in there ;). Maybe actually look at that engine and make sure it does what it advertises to be doing

### PR DESCRIPTION
**Pull request by mipek**
Not 100% sure if this is the way crit is supposed to work.
I translated the code to transitional syntax while at it.

This should fix #27

### CODE REVIEW

File: addons/sourcemod/scripting/WCX_Engine_Crit.sp [hunk 1]
```diff
@@ -8,30 +8,30 @@ public Plugin:myinfo =
     description="Generic crit skill"
 };
 
-public OnWar3EventPostHurt(victim, attacker, Float:damage, const String:weapon[32], bool:isWarcraft)
+public int OnWar3EventPostHurt(int victim, int attacker, float damage, const char weapon[32], bool isWarcraft)
```
Comment (derekl7) [inline addons/sourcemod/scripting/WCX_Engine_Crit.sp]: what does return give you?

Comment (derekl7) [inline addons/sourcemod/scripting/WCX_Engine_Crit.sp]: this is source pawn, why are these changed to floats? is this a new sourcemod compiler?

Comment (DonRevan) [inline addons/sourcemod/scripting/WCX_Engine_Crit.sp]: SourcePawn always had a (implicit) integer return value(prior to new syntax) which simply returned zero.
If I declare _OnWar3EventPostHurt_ as void the compiler complains about a prototype mismatch(because the forward is in old syntax -> int return value). Easy fix is to set return value as integer and return 0 to replicate old behavior.

**Comment by derekl7** [comment]
so ill have to update compiler on continuous build?

**Comment by DonRevan** [comment]
The compiler is already up2date. This is not the first transitional syntax code in War3Source btw.
Buildbot compiled fine: http://ownageclan.com/jenkins/job/War3Source-Default1/331/console

**Comment by derekl7** [comment]
that only builds the master branch i think, which is kind of misleading when build succeeds

ill merge and let it build master
